src/string.cr

```diff
@@ -3350,23 +3355,45 @@ class String
     nil
   end
 
+  private def index_short(hash_type, offset : Int32, pointer : UInt8*, end_pointer : UInt8*, search, &)
+    search_hash = hash_type.new(0)
+    hash = hash_type.new(0)
+    mask = hash_type.new(0)
+
+    search.each_byte do |b|
+      search_hash = (search_hash << 8) | b
+      hash = (hash << 8) | pointer.value
+      mask = (mask << 8) | 0xff
+      pointer += 1
+    end
+
+    while true
+      return offset if (hash & mask) == search_hash
+
+      char_bytesize = yield pointer
+      return if pointer + char_bytesize > end_pointer
+      case char_bytesize
+      when 1 then update_simplehash 1
+      when 2 then update_simplehash 2
+      when 3 then update_simplehash 3
+      else        update_simplehash 4
+      end
```

Comment:
```suggestion
      update_simplehash(char_bytesize.clamp(1, 4))
```

Reply:
`update_simplehash` is a macro that accepts a number literal and explicitly unrolls the loop, so `clamp` won't work here